CharlieBackend.Api/Middlewares/ExceptionHandleMiddleware.cs

```diff
@@ -44,7 +45,21 @@ public async Task InvokeAsync(HttpContext context)
         private Task HandleExceptionAsync(HttpContext context, Exception exception)
         {
             context.Response.ContentType = "application/json";
-            context.Response.StatusCode = 500;
+
+            switch(exception.GetType().Name)
+            {
+                case nameof(EntityValidationException):
+                    context.Response.StatusCode = 400;
+                    break;
+                case nameof(NotFoundException):
+                    context.Response.StatusCode = 404;
+                    break;
+                case nameof(Exception):
```

Comment:
Make default to return 500 error. Don't rethrow error here

Reply:
Done